checkita-core/src/main/scala/ru/raiffeisen/checkita/connections/greenplum/PivotalConnection.scala

```diff
@@ -0,0 +1,76 @@
+package ru.raiffeisen.checkita.connections.greenplum
+
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+import ru.raiffeisen.checkita.appsettings.AppSettings
+import ru.raiffeisen.checkita.config.jobconf.Sources.GreenplumSourceConfig
+import ru.raiffeisen.checkita.connections.DQConnection
+import ru.raiffeisen.checkita.config.jobconf.Connections.GreenplumConnectionConfig
+import ru.raiffeisen.checkita.readers.SchemaReaders.SourceSchema
+import ru.raiffeisen.checkita.utils.Common.paramsSeqToMap
+import ru.raiffeisen.checkita.utils.ResultUtils._
+
+import java.util.Properties
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+
+
+/**
+ * Connection to greenplum database
+ *
+ * @param config Connection configuration
+ */
+case class PivotalConnection(config: GreenplumConnectionConfig) extends DQConnection
+{
+  type SourceType = GreenplumSourceConfig
+
+  val id: String = config.id.value
+  protected val sparkParams: Seq[String] = config.parameters.map(_.value)
+  protected val connectionUrl: String =  config.url.value
+  protected val currentSchema: Option[String] = config.schema.map(_.value)
+
+  /**
+   * Gets basic greenplum connection properties
+   *
+   * @return Connection properties
+   */
+  def getProperties: Properties = {
+    val props = new Properties()
+    config.username.map(_.value).foreach(props.put("user", _))
+    config.password.map(_.value).foreach(props.put("password", _))
+    props.put("url", connectionUrl)
+    props.put("dbschema", config.dbschema.get.value)
+    props.put("dbtable", config.dbtable.get.value)
+    props
+  }
+
+  /**
+   * Checks connection.
+   *
+   * @param spark Implicit spark session object
+   * @return Nothing or error message in case if connection is not ready.
+   */
+  def checkConnection(implicit spark: SparkSession): Result[Unit] = Try {
+    val df = spark.read.format("greenplum").options(getProperties.asScala).load()
```

Comment:
Lets establish JDBC connection here using PostresSQL driver instead of loading spark dataframe. The goal here is to check that database is accessible and that provided credentials are valid.